Source/DafnyCore/DafnyPrelude.bpl

```diff
@@ -653,164 +653,164 @@ procedure $IterCollectNewObjects(prevHeap: Heap, newHeap: Heap, this: ref, NW: F
 
-function Set#Difference<T>(Set T, Set T): Set T;
-axiom (forall<T> a: Set T, b: Set T, o: T :: { Set#Difference(a,b)[o] }
+function Set#Difference(Set, Set): Set;
+axiom (forall a: Set, b: Set, o: Box :: { Set#Difference(a,b)[o] }
   Set#Difference(a,b)[o] <==> a[o] && !b[o]);
-axiom (forall<T> a, b: Set T, y: T :: { Set#Difference(a, b), b[y] }
+axiom (forall a, b: Set, y: Box :: { Set#Difference(a, b), b[y] }
   b[y] ==> !Set#Difference(a, b)[y] );
-axiom (forall<T> a, b: Set T ::
+axiom (forall a, b: Set ::
   { Set#Card(Set#Difference(a, b)) }
   Set#Card(Set#Difference(a, b)) + Set#Card(Set#Difference(b, a))
   + Set#Card(Set#Intersection(a, b))
     == Set#Card(Set#Union(a, b)) &&
   Set#Card(Set#Difference(a, b)) == Set#Card(a) - Set#Card(Set#Intersection(a, b)));
 
-function Set#Subset<T>(Set T, Set T): bool;
-axiom (forall<T> a: Set T, b: Set T :: { Set#Subset(a,b) }
-  Set#Subset(a,b) <==> (forall o: T :: {a[o]} {b[o]} a[o] ==> b[o]));
-// axiom(forall<T> a: Set T, b: Set T ::
+function Set#Subset(Set, Set): bool;
+axiom (forall a: Set, b: Set :: { Set#Subset(a,b) }
+  Set#Subset(a,b) <==> (forall o: Box :: {a[o]} {b[o]} a[o] ==> b[o]));
+// axiom(forall a: Set, b: Set ::
 //   { Set#Subset(a,b), Set#Card(a), Set#Card(b) }  // very restrictive trigger
 //   Set#Subset(a,b) ==> Set#Card(a) <= Set#Card(b));
 
 
-function Set#Equal<T>(Set T, Set T): bool;
-axiom (forall<T> a: Set T, b: Set T :: { Set#Equal(a,b) }
-  Set#Equal(a,b) <==> (forall o: T :: {a[o]} {b[o]} a[o] <==> b[o]));
-axiom (forall<T> a: Set T, b: Set T :: { Set#Equal(a,b) }  // extensionality axiom for sets
+function Set#Equal(Set, Set): bool;
+axiom (forall a: Set, b: Set :: { Set#Equal(a,b) }
+  Set#Equal(a,b) <==> (forall o: Box :: {a[o]} {b[o]} a[o] <==> b[o]));
+axiom (forall a: Set, b: Set :: { Set#Equal(a,b) }  // extensionality axiom for sets
   Set#Equal(a,b) ==> a == b);
 
-function Set#Disjoint<T>(Set T, Set T): bool;
-axiom (forall<T> a: Set T, b: Set T :: { Set#Disjoint(a,b) }
-  Set#Disjoint(a,b) <==> (forall o: T :: {a[o]} {b[o]} !a[o] || !b[o]));
+function Set#Disjoint(Set, Set): bool;
+axiom (forall a: Set, b: Set :: { Set#Disjoint(a,b) }
+  Set#Disjoint(a,b) <==> (forall o: Box :: {a[o]} {b[o]} !a[o] || !b[o]));
 
 // ---------------------------------------------------------------
 // -- Axiomatization of isets -------------------------------------
 // ---------------------------------------------------------------
 
-type ISet T = [T]bool;
+type ISet = [Box]bool;
 
-function ISet#Empty<T>(): Set T;
-axiom (forall<T> o: T :: { ISet#Empty()[o] } !ISet#Empty()[o]);
+function ISet#Empty(): Set;
+axiom (forall o: Box :: { ISet#Empty()[o] } !ISet#Empty()[o]);
 
 // the empty set could be of anything
-//axiom (forall<T> t: Ty :: { $Is(ISet#Empty() : [T]bool, TISet(t)) } $Is(ISet#Empty() : [T]bool, TISet(t)));
+//axiom (forall t: Ty :: { $Is(ISet#Empty() : [Box]bool, TISet(t)) } $Is(ISet#Empty() : [Box]bool, TISet(t)));
```

Comment:
Likewise

Reply:
Addressed this in [this commit](https://github.com/dafny-lang/dafny/pull/4535/commits/013ffdbc4e3413ecf5e09b10a5d2a4d25d02ac4d).